resources/elk6/init_kibana.sh

```diff
@@ -2,14 +2,28 @@
 
 #kibana_url="localhost:5601"
 kibana_url="kibana.local:5601"
-add_saved_objects="curl -u elastic:changeme -k -XPOST 'http://"$kibana_url"/api/saved_objects/_bulk_create' -H 'Content-Type: application/json' -H \"kbn-xsrf: true\" -d @"
+elasticsearch_url="elasticsearch.local:9200"
+add_saved_objects="curl -s -u elastic:changeme -k -XPOST 'http://"$kibana_url"/api/saved_objects/_bulk_create' -H 'Content-Type: application/json' -H \"kbn-xsrf: true\" -d @"
 
 #Create all saved objects - including index pattern
 saved_objects_file="kibana_APIonly.json"
 
 #if [ `curl -I localhost:5601/status | head -n1 |cut -d$' ' -f2` -eq '200' ]; then echo "Loading VulnWhisperer Saved Objects"; eval $(echo $add_saved_objects$saved_objects_file); else echo "waiting for kibana"; fi
- 
-until [ "`curl -I "$kibana_url"/status | head -n1 |cut -d$' ' -f2`" == "200" ]; do
+
+until curl -s "$elasticsearch_url/_cluster/health?pretty" | grep '"status"' | grep -qE "green|yellow"; do
+    curl -s "$elasticsearch_url/_cluster/health?pretty"
+    echo "Waiting for Elasticsearch"
+    sleep 5
+done
+
+echo "Loading VulnWhisperer index template"
+if curl -s --fail -XPUT "http://$elasticsearch_url/_template/vulnwhisperer" -H 'Content-Type: application/json' -d '@/opt/index-template.json'; then
+    echo -e "\nVulnWhisperer index template loaded successfully!"
+else
+    echo -e "\nFAILED to load VulnWhisperer index template"
```

Comment:
What would be the worst case scenario regarding the set up if the ElasticSearch template loads to fail and the rest of the script continues?

Reply:
The Kibana saved objects rely on a scan_fingerprint scripted field that requires `doc['asset.keyword']+'_'+doc['plugin_id']`.
Without the mapping the dashboards, and even the discover screen will fail to display with a big red message like `Error with Painless scripted field 'doc['asset.keyword']+'_'+doc['plugin_id']'.`

This is likely something we can move into the vulnerability standard.

Reply:
If we could move it to the vulnerability standard, it would be great, lets go for that then!